src/components/wizards/create-vm/CreateVMWizard.js

```diff
@@ -0,0 +1,366 @@
+        }
+      }
+      if(basicSettings.namespace){
+        vm.metadata.namespace = basicSettings.namespace.value;
+      }
+      if(basicSettings.description){
+        vm.metadata.labels.description = basicSettings.description;
+      }
+      if(basicSettings.flavor.value === 'Custom'){
+        vm.spec.template.spec = {
+          domain: {
+            cpu: {
+              cores: basicSettings.cpu.value
+            },
+            resources: {
+              requests: {
+                memory: `${basicSettings.memory.value}Gi`
+              }
+            }
+          }
+        }
+      } else {
+        vm.spec.template.spec = {
+          metadata: {
+            labels: {
+              'kubevirt.io/flavor': basicSettings.flavor.value
+            }
+          }
+        }
+      }
+      if(_.get(basicSettings.imageSourceType, 'value') === 'Registry'){
+        let domain = _.get(vm.spec.template.spec, 'domain', {});
+        let devices = _.get(domain,'devices',{});
+        let disks = _.get(devices,'disks',[]);
+        disks.push({
+          name: 'registryvolume',
+          volumeName: 'registryvolume',
+          disk: {
+            bus: 'virtio'
+          }
+        });
+        devices.disks = disks;
+        domain.devices = devices;
+        vm.spec.template.spec.domain = domain;
+
+        let volumes = _.get(vm.spec.template.spec, 'volumes', []);
+        volumes.push({
+          name: 'registryvolume',
+          registryDisk: {
+            image: basicSettings.registryImage.value
+          }
+        });
+        vm.spec.template.spec.volumes = volumes;
+      }
+      if(_.get(basicSettings.cloudInit, 'value', false)) {
+        let domain = _.get(vm.spec.template.spec, 'domain', {});
+        let devices = _.get(domain,'devices',[]);
+        devices.push({
+          name: 'cloudinitdisk',
+          volumeName: 'cloudinitdisk',
+          disk: {
+            bus: 'virtio'
+          }
+        });
+        domain.devices = devices;
+        vm.spec.template.spec.domain = domain;
+
+        let volumes = _.get(vm.spec.template.spec, 'volumes', []);
+        volumes.push({
+          name: 'cloudinitvolume',
+          cloudInitNoCloud: {
+            userData: {
+              'ssh-authorized-keys': [
+                `${basicSettings.authKeys.value} ${basicSettings.hostName.value}`
+              ]
+            }
+          }
+        });
+        vm.spec.template.spec.volumes = volumes;
+      }
+      if(basicSettings.startVM){
+        vm.spec.running=basicSettings.startVM.value;
+      }
+      return vm;
+    }
+  
+    closeWizard = () => {
+      this.setState(defaultState);
+      this.props.closeWizard();
+    }
+
+    onStepChanged = (index) => {
+      this.setState({ activeStepIndex: index })
+    }
+  
+    render() {
+      return <Wizard.Pattern
+        show={this.props.show}
+        onHide={this.closeWizard}
+        onClose={this.closeWizard}
+        steps={this.wizardStepsNewVM}
```

Comment:
`steps` is where the vast majority of wizard UI & logic lives.

I think that in future, we should find a better representation, I recall talking with patternfly-react folks about allowing `steps` to be represented as React components, instead of plain JS array.